exercises/rest-api/canonical-data.json

```diff
@@ -0,0 +1,292 @@
+{
+    "exercise": "rest-api",
+    "version": "1.0.0",
+    "comments": [
+      "The state of the API database before the request is represented",
+      "by the input->database object. Your track may determine how this",
+      "initial state is set",
+      "The inputs and outputs are represented as arrays of strings to",
+      "improve readability in this JSON file.",
+      "Your track may choose whether to present the input as a single",
+      "string (concatenating all the lines) or as the list.",
+      "In most cases, it seems to make sense to expect the output as",
+      "a single string.",
+      "Additionally, your track may choose whether to enforce or ignore",
+      "whitespace in JSON strings."
+    ],
+    "cases": [
+        {
+            "description": "user management",
+            "cases": [
+                {
+                    "description": "no users",
+                    "property": "request",
+                    "input": {
+                        "database": {
+                            "users": []
+                        },
+                        "request_body": ["GET /users"]
```

Comment:
I was wondering if maybe making this an object would be useful:

```json
"request_body": {
  "method": "GET",
  "url": "/users"
}
```

For requests with a payload, we could then add a "payload" field.

Reply:
I thought about this, and I suppose it comes down to the question: what is the real focus of this exercise? String parsing? Data management?

If we are going to extract method and URL, I suggest splitting the `request` property into `get` and `post`.

Reply:
I feel the real focus is in providing a REST API over some data. That this will involve some parsing and data management is important, but not the main focus. To me, it would make sense to have the canonical data align more with the HTTP/REST part than the data management part. But that is of course personal preference.